README.md

```diff
@@ -107,6 +107,35 @@ end
 
 Note that it is set to a string of the class constant, and not an instance of the class. The system constantizes and initializes a new instance whenever it uses the gateway to ensure autoloading works correctly and so a gateway that cares about the current state of the application is applied correctly. This is most relevant in multi-site environments where each site might want to use their own gift card gateway.
 
+### Testing Your Custom Gateway
+
+Workarea comes with unit tests for the main Gateway class, and you can
+decorate this test to provide your own setup code. A minimal
+implementation would look as follows:
+
+```ruby
+module Workarea
+  module GiftCards
+    decorate GatewayTest do
+      # This method wraps each test in a differently-named VCR cassette
+      # so you can re-use the same test logic without having to override
+      # every test. All you need to do is override this method to
+      # save off external API calls in each test, as by default they do not use
+      # a VCR cassette since they don't make external API calls.
+      def with_cassette(name)
+        VCR.use_cassette "custom_gift_card/#{name}" do
```

Comment:
Maybe we could just make this the implementation of this method out of the box by making this more specific. e.g.

```suggestion
        VCR.use_cassette "#{gateway.class.name.downcase.gsub('::', '_')}/#{name}" do
```

Reply:
If we did this, they probably wouldn't even have to touch this method then right? Just override the `gateway` and off to the races.

Reply:
i am all about that. similar to how credit card integration tests work.